azure/iac/sql_alert_policy.rego

```diff
@@ -1,91 +1,141 @@
-    "Policy Title": "",
-    "Policy Description": "",
+    "Policy Title": "Advanced data security should be enabled on your SQL servers.",
+    "Policy Description": "Advanced data security should be enabled on your SQL servers.",
     "Resource Type": "securityalertpolicies",
     "Policy Help URL": "",
     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/servers/securityalertpolicies"
 }
 
+# PR-AZR-0103-ARM
 
 default sql_managed_instance_alert = null
 
+azure_attribute_absence["sql_instance_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/managedinstances"
+    sql_resources := resource.resources[_]
+    lower(sql_resources.type) == "securityalertpolicies"
+    not sql_resources.properties.state
+}
+
+azure_attribute_absence["sql_instance_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/managedinstances/securityalertpolicies"
+    not resource.properties.state
+}
+
 azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/managedinstances"
     sql_resources := resource.resources[_]
     lower(sql_resources.type) == "securityalertpolicies"
-    sql_resources.properties.state == "Disabled"
+    lower(sql_resources.properties.state) == "disabled"
 }
 
 azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/managedinstances/securityalertpolicies"
-    resource.properties.state == "Disabled"
+    lower(resource.properties.state) == "disabled"
 }
 
 sql_managed_instance_alert {
     lower(input.resources[_].type) == "microsoft.sql/managedinstances"
+    not azure_attribute_absence["sql_instance_security_alert_disabled"]
     not azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"]
 }
 
+sql_managed_instance_alert = false {
+    azure_attribute_absence["sql_instance_security_alert_disabled"]
+}
+
 sql_managed_instance_alert = false {
     azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"]
 }
 
+sql_managed_instance_alert_err = "Security alert for SQL server missing in the resource!" {
+    azure_attribute_absence["sql_instance_security_alert_disabled"]
+}
+
 sql_managed_instance_alert_err = "Security alert for SQL managed instance is Disabled" {
     azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"]
 }
 
 sql_managed_instance_alert_metadata := {
-    "Policy Code": "",
+    "Policy Code": "PR-AZR-0103-ARM",
     "Type": "IaC",
     "Product": "AZR",
     "Language": "ARM template",
-    "Policy Title": "Security alert for SQL managed instance is Disabled",
-    "Policy Description": "Security alert for SQL managed instance is Disabled",
+    "Policy Title": "Advanced data security should be enabled on your SQL managed instance.",
```

Comment:
Advanced data security is a big topic but as per rule this Policy Title should be: `"Ensure Security Alert is enabled on Azure SQL Managed Instance"`

Reply:
fixed